### DISCUSSION TITLE
Connection issue with Strongswan PSK to Debian 9 racoon v0.8.2

### DISCUSSION THREAD

**Discussion by tomglx** [strongswan/strongswan#1641] [Q&A]
This is strongswan v5.9.10 on Fedora 37.
Hi, and here's another simple connection which worked flawlessly with racoon v0.8.2. But with strongswan it fails in Phase 1 (IKE).
No Idea why, because I can't any clear error message, other than a timeout. AFAIK there's also a racoon on Debian 9 on the other
side (0.8.2+20140711-8+deb9u1).
Here is the excerpt from swanctl.conf and a connection log file.

connections {
    conn1 {
        version = 1
        local_addrs = 1.1.1.1
        remote_addrs = 1.1.1.2
        proposals = aes256-sha1-modp1024, default
        rekey_time = 8h
        reauth_time = 0
        dpd_delay   = 30s
        dpd_timeout = 120s
        encap = yes
        mobike = no
        local {
            auth = psk
        }
        remote {
            auth = psk
            id = 1.1.1.2
        }
        children {
            sa1 {
                esp_proposals = aes256-sha1-modp1024, default
                local_ts = 10.0.0.0/8
                remote_ts = 1.1.1.2/32
                rekey_time = 8h
                mode = tunnel
                start_action = start
                dpd_action = restart
                policies_fwd_out = yes
                updown = "/etc/strongswan/updown.sh"
            }
            sa2 {
                esp_proposals = aes256-sha1-modp1024, default
                local_ts = 172.16.128.0/19
                remote_ts = 1.1.1.2/32
                rekey_time = 8h
                mode = tunnel
                dpd_action = restart
                policies_fwd_out = yes
                updown = "/etc/strongswan/updown.sh"
            }
        }
    }
}
secrets {
    ike-conn1 {
        secret = "xxx"
        id = 1.1.1.2
    }
}

[mssc.log](https://github.com/strongswan/strongswan/files/11146616/mssc.log)

**Comment by tobiasbrunner** [comment]
You have to check the log at the remote end (and whether it actually receives any messages). Because no response to the first message could be a lot of things (e.g. wrong remote IP, routing or firewall issue, no daemon running at the remote end, wrong IKE version, no matching config loaded, issues with the proposal selection and suppressed error reporting etc.).

**Comment by tomglx** [comment]
strongswan with default loglevel = 4 has this last messages. All with the original IP Adresses

Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[NET] received packet: from 217.91.199.39[500] to 194.39.189.12[500]
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC] parsing header of message
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC] parsing HEADER payload, 76 bytes left
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC] parsing payload from => 76 bytes @ 0x7fbe700051c0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    0: E1 2F 0C 34 2B 7E B9 EE CA 4C E4 F5 C2 74 98 7D  ./.4+~...L...t.}
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   16: 05 10 02 01 00 00 00 00 00 00 00 4C 87 35 71 0F  ...........L.5q.
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   32: 19 5C AC BB 3F 82 CF 62 EF E7 CA 9C 48 CB 85 F1  .\..?..b....H...
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   48: 5D 6F C3 AE 1D 5A 62 43 61 63 B3 87 A4 3D 5F 3F  ]o...ZbCac...=_?
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   64: 3D E9 29 CA 5E FB 60 E3 32 AF CE 9F              =.).^.`.2...
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 0 IKE_SPI
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 8 bytes @ 0x7fbe70017cc8
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    0: E1 2F 0C 34 2B 7E B9 EE                          ./.4+~..
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 1 IKE_SPI
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 8 bytes @ 0x7fbe70017cd0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    0: CA 4C E4 F5 C2 74 98 7D                          .L...t.}
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 2 U_INT_8
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 5
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 3 U_INT_4
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 1
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 4 U_INT_4
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 5 U_INT_8
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 2
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 6 RESERVED_BIT
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 7 RESERVED_BIT
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 8 FLAG
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 9 FLAG
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 10 FLAG
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 11 FLAG
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 12 FLAG
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 13 FLAG
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 1
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 14 U_INT_32
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 0
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]   parsing rule 15 HEADER_LENGTH
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC]    => 76
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC] parsing HEADER payload finished
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[ENC] parsed a ID_PROT message header
Apr  4 15:48:58 cvk-fw2 charon[15037]: 02[NET] waiting for data on sockets
Apr  4 15:48:58 cvk-fw2 charon[15037]: 10[MGR] checkout IKEv1 SA by message with SPIs e12f0c342b7eb9ee_i ca4ce4f5c274987d_r
Apr  4 15:48:58 cvk-fw2 charon[15037]: 10[MGR] IKE_SA checkout not successful

##
## Phase 1
##
remote 194.39.189.12 {
        exchange_mode main;
        lifetime time 8 hour;
        proposal_check claim;
        my_identifier address 194.39.189.12;
        ## TODO:
        nat_traversal off;
        proposal {
                encryption_algorithm aes 256;
                hash_algorithm sha1;
                authentication_method pre_shared_key;
                dh_group 2;
        }                                                                                                                     
}

##
## Phase 2
##

sainfo address 217.91.199.39/32 any  address 10.0.0.0/8 any {
        pfs_group 2;
        lifetime time 8 hour;
        encryption_algorithm aes 256;
        authentication_algorithm hmac_sha1;
        compression_algorithm deflate;
}
[racoon.log](https://github.com/strongswan/strongswan/files/11149303/racoon.log)

**Reply by tobiasbrunner** [reply]
I don't see log messages in the racoon log that correspond to the log excerpt you posted above (that remote SPI is not seen in the racoon log). So no idea if that's a response that's delayed or what packet that actually is.  I also don't see a `... bytes message received from ...` message in the racoon log that would correspond to the 260 byte sized message sent by strongSwan in the original log. Try to get logs that correspond exactly.

**Comment by tomglx** [comment]
The real remote ip is 217.91.199.39. Our GW has 194.39.189.12.
The reason that there a no viable log data from the remote racoon is, that it logs exactly nothing of the strongswan requests.
When racoons run on both sides it does. So I assume it doesn't even recognize the ISAKMP packets from strongswan and sends
ESP packets instead, because it believes that the SAs made with the previously running racoon on our side are still intact.
For completeness I've attached still logs of both sides.
[local.log](https://github.com/strongswan/strongswan/files/11157110/local.log)
[racoon.log](https://github.com/strongswan/strongswan/files/11157112/racoon.log)

**Comment by tomglx** [comment]
I found an issue with the calculated Phase 1 rekey_time. Although we set 8h in our strongswan configuration, it sends a value of 31680 seconds in the phase 1 proposal to racoon. 8h should result in 28800s. racoon compares it's own setting of 28800s to the time sent
in the proposal and rejects it, because the values doesn't match. To mitigate this, we tried to set over_time = 0. But this seems not to
work all of the time. 
What can we do, to ensure that the configured rekey time is included in strongswans Phase 1 proposal?

**Reply by tobiasbrunner** [reply]
> Although we set 8h in our strongswan configuration, it sends a value of 31680 seconds in the phase 1 proposal to racoon.  8h should result in 28800s.

strongSwan sends a calculated hard lifetime, which, as you noticed, is based on `rekey_time` and `over_time` but also `rand_time`, see [the docs](https://docs.strongswan.org/docs/5.9/config/rekeying.html) for details.

> racoon compares it's own setting of 28800s to the time sent
> in the proposal and rejects it, because the values doesn't match.

Doesn't make sense, but that's IKEv1 for you (note that the behavior is up to the implementation according to [RFC 2407, section 4.5.4](https://datatracker.ietf.org/doc/html/rfc2407#section-4.5.4)). Maybe there is an option for racoon to ignore a mismatch.

> To mitigate this, we tried to set over_time = 0. But this seems not to
> work all of the time.

That's a bad idea as the SA will expire at the same time a reauth/rekeying is triggered, the delete job could get executed before the reauth/rekey job.



> What can we do, to ensure that the configured rekey time is included in strongswans Phase 1 proposal?

Just configure the values so that it results in exactly `28800s` with some time to spare for the actual reauth to complete, e.g. `reauth_time = 28500s`, `over_time = 300s` and `rand_time = 0`.
